prov/shm/src/smr.h

```diff
@@ -184,6 +185,9 @@ static inline const char *smr_no_prefix(const char *addr)
 #define smr_fast_rma_enabled(mode, order) (!smr_env.disable_cma && \
 			(mode & FI_MR_VIRT_ADDR) && \
 			!(order & SMR_RMA_ORDER))
+#define smr_get_offset(base, addr) (uintptr_t)((char *) addr - \
+				    (char *) base)
+#define smr_get_addr(base, offset) ((char *) base + (uintptr_t) offset)
```

Comment:
Can you separate this into its own patch and update the other instances of this? (format_inject, format_iov, progress_inject, progress_iov, progress_resp_entry, etc)

Reply:
Done.